Lib/test/test_typing.py

```diff
@@ -4294,6 +4294,18 @@ def test_get_type_hints(self):
             {'a': typing.Optional[int], 'b': int}
         )
 
+    def test_custom_metaclass(self):
+        class _Foo(TypedDict):
+            pass
+
+        class _CustomTypedDictMeta(type(_Foo)):
+            pass
+
+        class _NewTypedDict(_Foo, metaclass=_CustomTypedDictMeta):
```

Comment:
Do we need an explicit test for case when we try to create a custom `_TypedDict` subclass that uses wrong metaclass? `class _NewTypedDict(_Foo, metaclass=Wrong):`

Reply:
In such case it will simply fail because of metaclass conflict:
```py
class _Foo(TypedDict):
    pass


class Wrong(type):
    pass


class _NewTypedDict(_Foo, metaclass=Wrong):
    pass
```
```py
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

I don't know if we should cover this case. I think it will be enough to verify that `TypedDict` subclass is an instance of a correct metaclass.